src/js/view/controls/controlbar.js

```diff
@@ -499,10 +519,17 @@ export default class Controlbar {
 
     updateButtons(model, newButtons = [], oldButtons = []) {
         const buttonContainer = this.elements.buttonContainer;
-        this.removeButtons(buttonContainer, oldButtons);
 
-        for (let i = newButtons.length - 1; i >= 0; i--) {
-            let buttonProps = newButtons[i];
+        const removedButtons = oldButtons.filter(oldButton =>
+            !newButtons.some(newButton => newButton.id + newButton.btnClass === oldButton.id + oldButton.btnClass));
```

Comment:
`includes` definitely wouldn't work here ?

Reply:
Includes is ES6 and I'm not sure how or if Babel would transform it https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#Specifications

Also the object instances are likely not the same, so comparing properties (`id` and `btnClass`) with `some` gets it done.

Reply:
```js
[3].includes(3)
// true

[{a: 3}].includes({a: 3})
// false

[{a: 3}].some(o => o.a === {a: 3}.a)
// true
```

Reply:
👍